### ISSUE TITLE
Add base of "search result" components

### ISSUE THREAD

**Issue by andrewtavis** [activist-org/activist#119]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

This is a simple issue to first add a baseline `SearchResult.vue` file that will house all the variations of the search results for events, organizations, resources and other types of page elements. This component can be found in the [Components tab of the designs](https://www.figma.com/file/I9McFfaLu1RiiWp5IP3YjE/activist_public_designs?type=design&node-id=1606%3A585&t=5CR5mIoy4P0OjwjS-1). This is a simple issue that will block others to set up the variants. ~~This should be a simple rectangular component with a background that is the `light-distinct` and `dark-distinct` colors and with borders that match `light-section-div` and `dark-section-div`. The corners should be rounded and there should be padding for elements that come within~~. The component should have `card-style` applied to it. There should further be some props in the `<script>`:

```ts
defineProps<{
  name?: string;
  private?: boolean;
  description: string;
  searchResultType: string;  // what we pass `organization`, `event`, etc to
}>();
```

We'll add more props as they're needed for the variants :) A further feature of this is that the element's width should change based on `sidebar.collapsed` and `sidebar.collapsedSwitch`. There should be padding to the right and the left padding should change based on these two features to then change the width. This component should be put on [activist.org/home/](https://activist.org/home/) so we can play around with it 😊

### Contribution

Happy to support someone who wants to work on this or will do it myself soon :)

**Comment by andrewtavis** [comment]
Will also need to be reactive to mobile like below:

![Mode=Light, Mobile=True, Type=Organization, Reduced=False](https://github.com/activist-org/activist/assets/24387426/5c772519-1d46-4931-aa96-7a3ef0ae1801)

**Comment by andrewtavis** [comment]
Thanks as always for the help, @momanyisamuel! Let me know if you need more details for what needs to be done :)

**Comment by andrewtavis** [comment]
@momanyisamuel, do you think it’s possible to get to this by Wednesday? We’ll be needing this to work on the mentioned issues during the hacking night we’ll be doing :) :)

**Comment by andrewtavis** [comment]
@momanyisamuel, I just made the custom Tailwind class `card-style` in #126 for this and other page elements, btw :)

**Comment by momanyisamuel** [comment]
Hey @andrewtavis please take a look at the component hope it can help with the meetup. sorry for the delay

**Comment by andrewtavis** [comment]
Legend 🙇‍♂️ Thanks @momanyisamuel! 😊😊 Will take a look and I’m sure it’ll be what we’re looking for :)

**Comment by andrewtavis** [comment]
And no stress on any delay whatsoever :) Really appreciate the quick response to all this! Let me know if you’d like to jump in on a call tomorrow as well :)

**Comment by momanyisamuel** [comment]
I wouldn't mind at all actually.

**Comment by andrewtavis** [comment]
I'll write into the Matrix chat tomorrow at 17:00 UTC and we can set something up 😊

**Comment by andrewtavis** [comment]
Closed via #130 🥳🙌
